### PR TITLE
increased left padding on TableCell and TableHeader for condensedStyl…

### PR DESCRIPTION
**Pull request by tranhoangan22**
Addresses #1618.

## Purpose

In the base table with sortable columns, the table header is too close to the sort-direction symbol. More left-padding needs to be added to the table headers and corresponding table cells.

## Approach and changes

- Increased left-padding on `TableHeader` and `TableCell` components (to `theme.spacings.giga`) for `condensedStyles` and `condensedPresentationStyles`
- Updated test snapshots 

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

## Visualisation of changes
Before
image
After
image

### CODE REVIEW

File: packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx [hunk 1]
```diff
@@ -161,7 +162,8 @@ const condensedColStyles = ({
   condensed &&
   scope === 'col' &&
   css`
-    padding: ${theme.spacings.byte} ${theme.spacings.mega};
+    padding: ${theme.spacings.kilo} ${theme.spacings.mega}
+      ${theme.spacings.kilo} ${theme.spacings.giga};
```
Comment (connor-baer) [inline packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx:166]: The vertical spacing was `byte` before, I assume this was an accidental change? 

```suggestion
    padding: ${theme.spacings.byte} ${theme.spacings.mega}
      ${theme.spacings.byte} ${theme.spacings.giga};
```

**Review by connor-baer — COMMENTED** [review]
Thank you for tackling this bug!

I think an unintended change snuck in (see my comment). Please don't forget to add a changeset :)

File: packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx (hunk 1 above)
Comment (tranhoangan22) [reply packages/circuit-ui/components/Table/components/TableHeader/TableHeader.tsx]: Thanks @connor-baer for spotting this! It was indeed an accidental change! I have fixed it, committed, added a changeset for this patch. Let me know if there's something else I missed out on!

File: .changeset/red-wolves-greet.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': patch
+---
+
+In the base table with sortable columns, the direction symbol was too close to the header text in condensed mode. This change increases left padding on TableCell and TableHeader for condensedStyle.
```
Comment (connor-baer) [inline .changeset/red-wolves-greet.md:5]: Changesets should be concise and focus on the _what_, not the _how_. Think about it this way: what information is relevant to a developer who is reading the changelog before upgrading Circuit UI in their application?

`TableCell` and `TableHeader` are internal components that the developer wouldn't know about. The padding is an implementation detail.

```suggestion
Fixed a spacing issue in the `Table` component when sorting a column in `condensed` mode.
```

Comment (tranhoangan22) [reply .changeset/red-wolves-greet.md]: Thanks for the nice comment! Makes a lot of sense! Committed the suggested change. :)

**Review by connor-baer — APPROVED** [review]
Woohoo, this is now ready to be merged! 🎉 Thank you for addressing all review comments.

**Pull request merged**